Uplift/Views/DropdownView.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  DropdownView.swift
+//  
+//
+//  Created by Cameron Hamidi on 10/7/19.
+//
+
+import Foundation
+import UIKit
+
+enum DropdownStatus {
+    case up, half, down
+}
+
+enum DropdownDirection {
+    case horizontal, vertical
+}
+
+protocol DropdownViewDelegate {
+    func collapseDropdownView(sender dropdownView: DropdownView)
+    func collapseDropdownViewHalf(sender dropdownView: DropdownView)
+    func expandDropdownViewFull(sender dropdownView: DropdownView)
+    func expandDropdownViewHalf(sender dropdownView: DropdownView)
+}
+
+class DropdownView: UIView {
+    var collapseHalfDropdownGesture: UITapGestureRecognizer!
+    var contentView: UIView!
+    var contentViewBottomConstraint: NSLayoutConstraint!
+    var contentViewHeight: CGFloat!
+    var contentViewHeightConstraint: NSLayoutConstraint!
+    var currentHeight: CGFloat!
+    var delegate: DropdownViewDelegate!
+    var expandCollapseDropdownGesture: UITapGestureRecognizer!
+    var expandFromHalfDropdownGesture: UITapGestureRecognizer!
+    var halfCollapseView: UIView?
+    var halfCollapseViewHeight: CGFloat!
+    var halfDropdownEnabled: Bool!
+    var halfExpandView: UIView?
+    var halfExpandViewHeight: CGFloat!
+    var halfHeight: CGFloat!
+    var headerView: UIView!
+    var headerViewHeight: CGFloat!
+    var status: DropdownStatus = .up
```

Comment:
can we make these `private // private let` where possible?

Reply:
same comment about making the names consistent to make the code more readable